vendor/engines/bulk_email/spec/services/bulk_email/recipient_searcher_spec.rb

```diff
@@ -402,6 +402,63 @@ def place_reservation(purchaser, at)
     end
   end
 
+  context "when searching for problem_reservation" do
+    let(:problem_reservation_users) { create_list(:user, 2) }
+    let(:user) { problem_reservation_users.first }
+    let(:user2) { problem_reservation_users.second }
+    let(:user3) { create(:user) }
+    let(:product) { create(:setup_instrument, :timer, :always_available, facility:, problems_resolvable_by_user: true) }
+    let(:product2) { create(:setup_instrument, :timer, :always_available, facility:, problems_resolvable_by_user: true, is_hidden: true) }
+    let(:product3) { create(:setup_instrument, :timer, :always_available, facility:, problems_resolvable_by_user: true, is_archived: true) }
+
+    # Users 1 and 2 have reservations with problems
+    # User 3 has a reservation without a problem
+    let!(:problem_reservation) { create(:purchased_reservation, product:, reserve_start_at: 2.hours.ago, reserve_end_at: 1.hour.ago, actual_start_at: 1.hour.ago, actual_end_at: nil, user:) }
+    let!(:problem_reservation2) { create(:purchased_reservation, product: product2, reserve_start_at: 2.hours.ago, reserve_end_at: 1.hour.ago, actual_start_at: 1.hour.ago, actual_end_at: nil, user: user2) }
+    let!(:problem_reservation3) { create(:purchased_reservation, product:, user: user3) }
```

Comment:
I think this should be named differently, since it is not a problem:
```suggestion
    let!(:reservation3) { create(:purchased_reservation, product:, user: user3) }
```